chore: date-service.coffee to TS

returnMarcco opened the pull request:
# Description

- This change migrates the `date-service.coffee` script to TypeScript.

Fixes # (issue)

https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.planner/tt.c_19:bd20175d09414f079490a2403f7fca74@thread.tacv2_p_njykIFLDn0iAY1at7tACfcgADgBS_h_1714385623719?tenantId=d02378ec-1688-46d5-8540-1c28b5f470f6&webUrl=https%3A%2F%2Ftasks.teams.microsoft.com%2Fteamsui%2FpersonalApp%2Falltasklists&context=%7B%22subEntityId%22%3A%22%2Fv1%2Fplan%2FnjykIFLDn0iAY1at7tACfcgADgBS%2Ftask%2Ftcw8YeDXvU-upRzKumYXsMgAD25D%22%2C%22channelId%22%3A%2219%3Abd20175d09414f079490a2403f7fca74%40thread.tacv2%22%7D

## Type of change

- [x] This change requires a documentation update (Requires this service to be added to the `done` section of the migration todos list within the codebase)
- [x] Frontend migration

# How Has This Been Tested?

- A unit test was written. I was not able to run the unit test for some reason, but ask the reviewer to attempt to run it if possible. Thanks.
- Manually tested in Chrome and Firefox (I do not have access to a Mac for Safari)
- Chrome Devtools `console` and `network` tabs were monitored - no obvious errors found
- I global-searched for references to the `date-service` and found ~22. The only components I could find that referenced this were the `Home` and `Unit Dropdown` components. I've triggered these components in the UI and found no obvious errors in `Devtools`

## Testing Checklist:

- [x] Tested in latest Chrome
- [ ] Tested in latest Safari
- [x] Tested in latest Firefox

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have requested a review from @macite and @jakerenzella on the Pull Request

returnMarcco committed:
chore: date-service.coffee to TS

- add `date.service.ts` file
- add `date.service.spec.ts` unit test
- downgrade TS module such that it works within AngularJS

b0ink commented:
Hey @returnMarcco 

The old coffeescript component still needs to be [unlinked](https://github.com/thoth-tech/doubtfire-web/blob/development/MIGRATION-GUIDE.md#step-3---unlink-old-component-and-add-in-new) from `src/app/doubtfire-angularjs.module.ts` by deleting the line that imports `date-service.js`, and removing `doubtfire.common.services.dates` from the `services.coffee` file

I was able to confirm your migration is working after unlinking the old component and rebuilding the app, and was able to verify the `showDate()` function works correctly in the unit card when viewing the `/Home` page

The old `date-service.coffee` file can also now be deleted

You can also import the new `DateService` class into the other TypeScript components that currently use the date service, and replace the `any` type with the `DateService` class to enforce type safety

returnMarcco commented:
@b0ink Thanks for the review. I'll make the changes and push asap. Do you want a component review for this, considering it's a service?

returnMarcco committed:
chore: unlink date-service.coffee

- complete unlinking process of  `date-service.coffee` after migrating to Typescript
- manual linting/comment formatting

returnMarcco committed:
chore: delete redundant coffeescript file

- delete `date-service.coffee` file after migration

returnMarcco commented:
@b0ink I've resolved the above requested changes.

returnMarcco commented:
I am awaiting 2x peer reviews before requesting another senior review from Eliya.

AB-Deakin commented:
![pull-rev-1](https://github.com/user-attachments/assets/62401488-5c4b-4ebc-b46a-5ed9ffa8fe42)
![pull-rev-2](https://github.com/user-attachments/assets/a1f6655c-69bb-449c-beb8-a358844c4d22)

AB-Deakin commented:
Completed peer review. Running correctly and tested in Chrome and Firefox (no access to Safari).

returnMarcco commented:
> Completed peer review. Running correctly and tested in Chrome and Firefox (no access to Safari).

Thanks for the peer review, much appreciated.

zara1504 commented:
Hi there, after testing your work in Google, your component ran correctly, and I didn't run into any issues. Along with this, after reading the comments above, I can see that you unlinked the component and removed it as well. Good job. :)

returnMarcco commented:
> Hi there, after testing your work in Google, your component ran correctly, and I didn't run into any issues. Along with this, after reading the comments above, I can see that you unlinked the component and removed it as well. Good job. :) 

Thanks @zara1504 , I appreciate it.

returnMarcco commented:
@b0ink Hi. I've just gotten 2x peer reviews and was asked by Nebula to request a second review from a leader. Cheers.

b0ink commented:
Looks good to me, can confirm the dates in the unit cards are displayed correctly, great job

![image](https://github.com/user-attachments/assets/9f0c62a1-cec4-4de1-8b66-f0fbe1a493ae)

One last change I can recommend is that you format your TypeScript files so that it's consistent with the rest of the codebase - make sure you format it with Prettier and not the default TypeScript formatter

You can do this quickly by opening the command line palette via CTRL+SHIFT+P, search for "Format Document With..." and then select Prettier

returnMarcco committed:
chore:  formatting

- format code with prettier

returnMarcco commented:
> Looks good to me, can confirm the dates in the unit cards are displayed correctly, great job
> 
> ![image](https://private-user-images.githubusercontent.com/40929320/429443194-9f0c62a1-cec4-4de1-8b66-f0fbe1a493ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM2Njk4NDQsIm5iZiI6MTc0MzY2OTU0NCwicGF0aCI6Ii80MDkyOTMyMC80Mjk0NDMxOTQtOWYwYzYyYTEtY2VjNC00ZGUxLThiNjYtZjBmYmUxYTQ5M2FlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDAzVDA4MzkwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1ZTY4NGNjMjYwYWFkMzBlZWRjNGNjYWRiYzJkODViOGRlNWUwOTI3Yjg1MzQ4OTMzYTk0MjFkMWRmYjI5MGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.71ayJ-UKbzrvBGDYV8X4CfP7daTq9IAIODaSDYPJaTI)
> 
> One last change I can recommend is that you format your TypeScript files so that it's consistent with the rest of the codebase - make sure you format it with Prettier and not the default TypeScript formatter
> 
> You can do this quickly by opening the command line palette via CTRL+SHIFT+P, search for "Format Document With..." and then select Prettier

Thanks for the feedback. I've now formatted the TS file with Prettier, commited the changes and pushed to origin.

AB-Deakin reviewed (approved):
Tested and working correctly. Peer Review approved by Alex Brown

aNebula reviewed (approved):
LGTM. Please open an upstream PR with these changes against [doubtfire-lms/doubtfire-web 8.0.x branch](https://github.com/doubtfire-lms/doubtfire-web/tree/8.0.x).
